src/test/java/seedu/bookmark/logic/commands/GoalDelCommandTest.java

```diff
@@ -0,0 +1,73 @@
+package seedu.bookmark.logic.commands;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.bookmark.logic.commands.CommandTestUtil.assertCommandFailure;
+import static seedu.bookmark.logic.commands.CommandTestUtil.assertCommandSuccess;
+import static seedu.bookmark.testutil.TypicalBooks.getTypicalLibrary;
+import static seedu.bookmark.testutil.TypicalIndexes.INDEX_FIRST_BOOK;
+import static seedu.bookmark.testutil.TypicalIndexes.INDEX_SECOND_BOOK;
+
+import org.junit.jupiter.api.Test;
+
+import seedu.bookmark.commons.core.Messages;
+import seedu.bookmark.commons.core.index.Index;
+import seedu.bookmark.model.Model;
+import seedu.bookmark.model.ModelManager;
+import seedu.bookmark.model.UserPrefs;
+import seedu.bookmark.model.book.Book;
+import seedu.bookmark.model.book.Goal;
+
+public class GoalDelCommandTest {
```

Comment:
Maybe one more test for if the book has no goals then you try to delete the goal?

Reply:
Thanks for the suggestion! Will add that in